pkg/api/customdata.go

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api
+
+import (
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"go.etcd.io/etcd/client/v3/concurrency"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	// CustomDataPrefix is the object prefix.
+	CustomDataPrefix = "/customdata/{kind}"
```

Comment:
Here `CustomDataPrefix = /customdata/{kind}` but in pkg/cluster/layout.go there is `customDataPrefixFormat = /custom-data/%`. Is it possible to have only one variable for this?

Reply:
unfortunately, we are not able to do this, the `{kind}` in `CustomDataPrefix` is for the `chi` framework to extract parameters in the request URL, '%s` in `customDataPrefixFormat` is for `fmt.Sprintf`.